File: ProjectsAPI/ProjectsAPI/Models/Project.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace ProjectsAPI.Models
{
    public class Project
    {
        public Project(string name, int allocation, DateTime startDate, DateTime endDate, string status)
        {
            this.Name = name;
            this.Allocation = allocation;
            this.StartDate = startDate;
            this.EndDate = endDate;
            this.Status = status;
        }

        public string Name { get; private set; }
        public int Allocation { get; private set; }
        public DateTime StartDate { get; private set; }
        public DateTime EndDate { get; private set; }
        public string Status { get; private set; }

        internal void Update(Project project)
        {
            this.Name = project.Name;
            this.Allocation = project.Allocation;
            this.StartDate = project.StartDate;
            this.EndDate = project.EndDate;
            this.Status = project.Status;
        }
    }
}

File: Projects_Jquery/JS/project.js
function Project() {
    this.name = "";
    this.allocation = 0;
    this.startdate = "";
    this.enddate = "";
    this.status = "Open";
}

Project.prototype.addProject = function() {
    this.name = $('#projectname').val();
    this.startdate = $('#startdate').val();
    this.enddate = $('#enddate').val();
    this.allocation = $('#allocation').val();
}

Project.prototype.changeStatus = function(status) {
    this.status = status;
}

Project.prototype.setProject = function(obj) {
    this.name = obj.name;
    this.allocation = obj.allocation;
    this.startdate = obj.startDate;
    this.enddate = obj.endDate;
    this.status = obj.status;
}